### PR TITLE
Remove the deprecated InfectionConfig class

### PR DESCRIPTION
**Pull request by theofidry**

### CODE REVIEW

File: src/Config/ValueProvider/TimeoutProvider.php [hunk 1]
```diff
@@ -1,84 +0,0 @@
-<?php
```
Comment (theofidry) [inline src/Config/ValueProvider/TimeoutProvider.php]: This class is no longer necessary: we pass the default timeout value (10s) in `ConfigurationFactory` and this setting was never required in `schema.json`.

**Comment by sanmai** [comment]
There's an E2E test failing for the `Configure` operation, do you think it might be related?

**Comment by theofidry** [comment]
Indeed it was related :)

**Comment by theofidry** [comment]
@maks-rafalko is there a way to configure the cov service? I'm not quite sure why it was failing in this PR but it looked wrong

**Comment by maks-rafalko** [comment]
what do you mean by configuring? the CodeCov config is located here:

https://github.com/infection/infection/blob/master/codecov.yml

**Pull request merged**